packages/istanbul-lib-coverage/lib/coverage-map.js

```diff
@@ -58,6 +58,19 @@ CoverageMap.prototype.merge = function (obj) {
     });
 };
 /**
+ * filter the coveragemap based on the callback provided
+ * @param {Function (filename)} callback - Returns true if the should be
```

Comment:
I think I'm blanking on how this additional filter option relates to the change in `test-exclude`, mind providing an example of the final functionality you're envisioning in the pull request description.

This is looking great btw, thanks for the contribution 👍